### DISCUSSION TITLE
Remove 63 chars in glyph names restriction?

### DISCUSSION THREAD

**Discussion by anthrotype** [adobe-type-tools/afdko#1706] [Ideas]
http://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#2fi-glyph-name

> A glyph name may be up to 63 characters in length

In 2015 this was lifted from 31 up to 63. https://github.com/adobe-type-tools/afdko/issues/70

in ufo2ft, some people have hit this 63 limit and aren't able to compile their fonts: https://github.com/googlefonts/ufo2ft/issues/588

fonttools feaLib compiler enforces this 63 chars limitation at parsing time, even though technically our python implementation doesn't use a fixed array to store these so it can parse without the check.

In https://github.com/fonttools/fonttools/issues/2530 I wondered if we just lift this restriction entirely, but I wanted to check with you first, to avoid implementations from diverging too much.

Do you require the names to be fixed length to avoid dynamically allocating memory? If not we can just get rid of the limit altogether perhaps.
Or how about we lift this again, say, to 127 characters?

**Comment by behdad** [comment]
cc @PeterCon

**Comment by josh-hadley** [comment]
My understanding of this limitation is that it comes from implementations (printer drivers and such); it's not simply an AFDKO/tool limitation. I can do some consultations internally and see whether the 63 character limit is still relevant.

**Comment by anthrotype** [comment]
Does makeotf actually enforce this at all? We're considering to drop the character limit altogether here: https://github.com/fonttools/fonttools/pull/3424
